### ISSUE TITLE
Please don't explicitly check for GCC

### ISSUE THREAD

**Issue by snej** [rudi-cilibrasi/zlibcomplete#4]
The following lines in zlibcomplete.hpp trigger a compilation failure when using any compiler other than GCC (i.e. Clang or MSVC):
```c++
#if CURRENT_GCC_VERSION < 40800
#error "Please use g++ version 4.8 or later."
#endif
```
Please don't do this. I want to use this library in a cross-platform product, so now I can't import it as a Git submodule; instead I have to copy the code or fork the repo, then modify it.

**Comment by rudi-cilibrasi** [comment]
Thanks for the bug report. I am glad to hear you are using it outside of gcc.
Does this commit fix it for you?

https://github.com/rudi-cilibrasi/zlibcomplete/commit/81c6e249a398b081ffe2aa44b872f5e661a5d9ae

**Comment by snej** [comment]
My apologies, I hadn't looked outside the context of that `#error`; there's a test for `#ifdef __GNUC__` surrounding it, which makes what you're doing more reasonable. However, Clang does define that symbol, and `__GNUC_MINOR__` and `__GNUC_PATCHLEVEL__`; the resulting `CURRENT_GCC_VERSION` is 40201, as of clang-802.0.35 (Xcode 8.3b3). I suppose Clang does this for compatibility reasons, because it also supports a number of GCC language extensions.

What's the reason behind this error? If you require certain language features it'd be better to test for them explicitly. Or is this to check for a GCC bug that was fixed in 4.8?

(Anyway, thanks for the quick fix!)

**Comment by rudi-cilibrasi** [comment]
:+1:
